File: integration/android/ProductDetailHostScreen.kt
import androidx.compose.runtime.Composable

class ProductDetailHostStore {
    val screenState = ProductDetailScreenState()

    fun addToCart() {}

    fun showShippingDetails() {}
}

@Composable
fun ProductDetailHostScreen(store: ProductDetailHostStore = ProductDetailHostStore()) {
    ProductDetailScreen(
        state = store.screenState,
        actions = ProductDetailScreenActions(
            onAddToCart = { store.addToCart() }
        ),
        navigation = ProductDetailScreenNavigation(
            onShippingDetails = { store.showShippingDetails() }
        )
    )
}

File: integration/ios/ProductDetailHostScreen.swift
import SwiftUI

final class ProductDetailHostStore {
    func addToCart() {}
    func showShippingDetails() {}
}

struct ProductDetailHostScreen: View {
    @State private var screenState = ProductDetailScreenState()
    private let store = ProductDetailHostStore()

    var body: some View {
        ProductDetailScreen(
            state: $screenState,
            actions: ProductDetailScreenActions(
                addToCart: { store.addToCart() }
            ),
            navigation: ProductDetailScreenNavigation(
                shippingDetails: { store.showShippingDetails() }
            )
        )
    }
}
